source/agora/consensus/PreImage.d

```diff
@@ -487,3 +490,93 @@ unittest
+    // (PreImageCycle.PreImageCount - 5040) times from the values of
+    // `hashMulti(secret, "consensus.preimages", 0)` for each secret.
+    if (pair == NODE7)
+        seed = Hash(`0xea0db105d9d18258fefa31a4bae2f8238199d9389d42aa29b2f26312e6d73b460d511a6d637f2c5a174441edb3176acbb8d18d6c7a0d790ef934111d36ff065f`);
+    else if (pair == NODE2)
+        seed = Hash(`0x83c41cc48d14bcc825dc69501f7d31a22d551fc8eac4e2d72a79e0cb7b027ec739db7e997c3e322e819d9073e5f71ebdca7ce6829471785b1d101522bf7f9946`);
+    else if (pair == NODE3)
+        seed = Hash(`0x84fd47fd68bc12d6a7c7e076fb261682db02b63c6d14c00753fae5894c26a42be1cb98b5a5e2a8c45f7037cfc3257aacba94b66e281a78407264ac57cd83a321`);
+    else if (pair == NODE6)
+        seed = Hash(`0x3c82790786faaeafab5c82c8cae4b459b445cd6698bf9ae372f24ed9994418a00d305598ef54fad8aaf9a732c148085372fb19bc8a6093c48b4e052fa8dcb579`);
+    else if (pair == NODE5)
+        seed = Hash(`0x2c9306c0262a50fc1ccaa7f3248f28ad6d6ecd0b195ca229d085cd3f3866fa7f618f8d5d9d3fc41d91cf6b6010a75a941c19d54548f1a8c78c070d48b0a94b50`);
+    else if (pair == NODE4)
+        seed = Hash(`0xcf650f6f81949783a789c8a117eb505f50cfe16bcefe6581bb9bba81ae26e540aade33855be49280f2e5287281e9d90525127eedae568d7b1773a374d5730fc7`);
+    else if (pair == A)
+        seed = Hash(`0xd19817b4270c891fb7f095d972de462389727306fb0af679bd1db336f16e096342e999124606da7951749491842d602b9fe64a38582aca3d730382d0be571b0a`);
+    else if (pair == C)
+        seed = Hash(`0x22d80a87812c42caf42f45ce0a8afe216066420abb465d4a679eb46ebadbac413a267482c42c3c6089763d417525aa6b5454a598f77d562270e855285c6369dd`);
+    else if (pair == D)
+        seed = Hash(`0x8f67adb05d8caffbe8e2bc5f852715bed171dcdedce3f1105f8cccb4ad4f2e200de407aebda6ec5882a47ca6911d734f5cbdcb947e6f4f766c599c9f76f333ce`);
+    else if (pair == E)
+        seed = Hash(`0x857365c0ba6215097e2b8c45195f4b17a4d69b3555f5eebca42ffa34254670c7edfbbc44b37491fc918e0ba0017afff453f679726e8832b15c4b24cc6fd413e8`);
+    else if (pair == F)
+        seed = Hash(`0xfba44700d3f22c19dd00e3b00853cb3b0e1805d91ddfbbe1820da69d36c7e40c471ed216dde85564ac2d226232dccbf0baca56fad0764904f181795972770936`);
+    else if (pair == G)
+        seed = Hash(`0xd244e2d05a4520799ba0d927e949087ef125524f1ba1b75d8d3b7e06983496a546b50ef5f2bac1b6db8f81dcadd30b87e67e67e46a42c863a4938c4485fafeeb`);
+    else if (pair == H)
+        seed = Hash(`0x0e2a80082b26abab2b5c1d6e7e6c15789d68c551c318d614ec4bb96e92bcec484ea13e50ff614e33f64c9fa2ff1783ebeb4d480cde7b3285a2475fc19e974532`);
+    else if (pair == J)
+        seed = Hash(`0x2f3273381c16e00509adbd4fb1871a1aa640b6c908ba5837775b02c9cec4e410f8057d0f7f2f742cbae73da9f3086d68cdb4daab54a0665651fcc57f74c6ec29`);
+    else if (pair == K)
+        seed = Hash(`0xc2929018e4cda8ce470d8f6a57052abe05ae4e095cd7f63b2654c7c457d3c085f110a1e8048646bef51cc3ef03442ce3452f267bfd92538b0dbfab0b479560f0`);
+    else
+    {
+        auto cycle = PreImageCycle(pair.secret, 10, PreImageCycle.PreImageCount, Height(0));
```

Comment:
So 10 here is the validator cycle? Maybe it is not needed in the `PreImageCycle` anymore as it only depends on the `height`.

Reply:
Yes, it's needed because we have seeds for all validator cycles. The validator cycle means the interval between the seeds. If it's not there, we have to have all the pre-images.

Reply:
Perhaps it can be renamed for the purpose it is used. Maybe `seed_interval` as it is longer coupled with the `enrolment` cycle` as far as I can tell.

Reply:
Yes. `seed_interval` could be better because `enrollment cycle` is not related to the interval anymore as you said. But the change of the name is not the main thing of this PR and we can think we use the validator cycle as seed interval. So let's leave it as it is.